### PR TITLE
Reduce amount of time the repository lock is held

### PR DESCRIPTION
**Pull request by fao89**
https://pulp.plan.io/issues/6677
closes #6677

### CODE REVIEW

File: pulp_ansible/app/tasks/collections.py [hunk 1]
```diff
@@ -195,6 +196,17 @@ def create_collection_from_importer(importer_result):
     return collection_version
 
 
+def add_collection_to_repository(repository_pk, collection_version_pk):
```
Comment (alikins) [inline pulp_ansible/app/tasks/collections.py:199]: This is presumably close enough to the same as https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/tasks/promotion.py#L4
that galaxy_ng should be able to use pulpcore.app.tasks.collections.add_collection_to_repository() once it is merged?

Comment (alikins) [reply pulp_ansible/app/tasks/collections.py:199]: pinging @awcrosby

Comment (awcrosby) [reply pulp_ansible/app/tasks/collections.py:199]: Yeah that is something we could look at replacing with this

File: pulp_ansible/app/galaxy/v3/views.py [hunk 2]
```diff
@@ -269,6 +269,15 @@ def retrieve(self, request, *args, **kwargs):
...
         instance = self.get_object()
+        task = instance.task
+
+        completed = ["completed", "failed", "canceled"]
+
+        for child in task.child_tasks.all():
+            if child.state not in completed:
+                task.state = child.state
+                task.finished_at = child.finished_at
+                break
```
Comment (fao89) [inline pulp_ansible/app/galaxy/v3/views.py:280]: @bmbouter this is what I mentioned felt like a hack

**Comment by fao89** [comment]
too old PR

**Pull request closed without merging**